td-agent/Rakefile

```diff
@@ -244,6 +244,15 @@ class BuildTask
     @download_task = download_task
   end
 
+  def make_wix_version_numer(version)
+    if version.split("~")[1]
+      version_base = version.split("~")[0].chars.delete_if{|e| e == "."}.join("").to_i.pred.to_s.chars.join(".")
+      version_base + "." + Time.now.strftime("%m%d")
```

Comment:
What about:

```
def make_wix_version_number(version)
  return version unless version.include?("~")
  "%s.%s" % [
    version.split("~", 2)[0].delete(".").to_i.pred.to_s.chars.join("."),
    Time.now.strftime("%y%m%d")
  ]
end
```

Reply:
Oops, need to remove "%y".

```
def make_wix_version_number(version)
  return version unless version.include?("~")
  "%s.%s" % [
    version.split("~", 2)[0].delete(".").to_i.pred.to_s.chars.join("."),
    Time.now.strftime("%m%d")
  ]
end
```

Reply:
Yep, Including `%y` should cause overflow for patch level version.
WiX version should be in 0 ~ 65535.